src/main/people.ts

```diff
@@ -0,0 +1,99 @@
+import { getBasecampUrl, sendPaginatedBasecampGetRequest } from "./basecamp";
+import { PersonMissingIdError } from "./error/personMissingIdError";
+import { PersonNameIdMapNotCachedError } from "./error/personNameIdMapNotCachedError";
+import { getScriptProperty, setScriptProperty } from "./propertiesService";
+
+type PersonNameIdMap = {[key: string]: string};
+
+// Project ID hardcoded to our SD Basecamp Integration project for testing
+// In the future we may consider moving this to a script property as a config value
+const PROJECT_ID: string = "38736474";
+const PEOPLE_PATH: string = `/projects/${PROJECT_ID}/people.json`;
+const PEOPLE_MAP_KEY: string = "PEOPLE_MAP";
+
+// Regex to match a person's name followed by a city in parentheses. ex. John MiddleName1 ... MiddleName5 Doe (SD)
+const CITY_REGEX: string = "^(.*)\(([^)]*)\)$";
+
+let cachedPersonNameIdMap: PersonNameIdMap | null = null;
+
+/**
+ * Utility function that retrieves people from the Basecamp API, selects SD people, and then populates
+ * the script properties with a map from person name to Basecamp person id. Can be invoked manually
+ * from the Google Apps Script Developer UI
+ */
+export function populatePeopleInDb(): void {
+    const requestUrl: string = getBasecampUrl() + PEOPLE_PATH;
+    const peopleData: Person[] = sendPaginatedBasecampGetRequest(requestUrl) as Person[];
+
+    // Reduce all of the people to a single map
+    const personNameIdMap: PersonNameIdMap = peopleData.reduce((map: PersonNameIdMap, person: Person) => {
+        const match: RegExpMatchArray | null = person.name.match(CITY_REGEX);
```

Comment:
small nitpick, the extracting of the person's name feels like it should be a separate function to maintain the same level of abstraction within the function